lib/internal/modules/cjs/loader.js

```diff
@@ -616,31 +633,33 @@ Module._load = function(request, parent, isMain) {
   }
 
   // Don't call updateChildren(), Module constructor already does.
-  var module = new Module(filename, parent);
+  const module = new Module(filename, parent);
 
   if (isMain) {
     process.mainModule = module;
     module.id = '.';
   }
 
   Module._cache[filename] = module;
+  if (parent !== undefined) {
+    relativeResolveCache[relResolveCacheIdentifier] = filename;
+  }
 
-  tryModuleLoad(module, filename);
-
-  return module.exports;
-};
-
-function tryModuleLoad(module, filename) {
```

Comment:
Did you test this particular change (or the entire PR for that matter) against node v8.x? I don't remember at what point try-catch/finally statements were more completely optimized in V8. I think node v10.x should be safe though. I'm just asking because currently there is no `do-not-land-on-v8.x` label for this PR.

Reply:
I did not benchmark this particular case but `try / catch` is optimized since V8 5.9 (https://benediktmeurer.de/2017/04/03/v8-behind-the-scenes-march-edition/) and Node.js v8 uses V8 6.2.